frontend/app/assets/javascripts/spree/frontend/cart.js

```diff
@@ -12,9 +12,21 @@ Spree.ready(function ($) {
       }
     }
 
-    formUpdateCart.find('a.delete').show().one('click', function () {
+    formUpdateCart.find('a.delete').show().one('click', function (event) {
+      var itemId = $(this).attr('data-id')
+      var link = $(event.currentTarget);
+      var quantityInputs = $("form#update-cart input.shopping-cart-item-quantity-input[data-id='" + itemId + "']")
+      var quantity = $(quantityInputs).val()
       $(this).parents('.shopping-cart-item').first().find('input.shopping-cart-item-quantity-input').val(0)
       clearInvalidCouponField()
+      link.trigger({
```

Comment:
@aplegatt let's please check if `link[0].dataset` is present before, also we need to check if `quantity` is present as well